File: mvc1/_etc/C#/1783 병든 나이트.cs
using System;
using static System.Console;


class Program
{
    static void Main() {
        string[] s = ReadLine().Split(' ');
        int n  = int.Parse(s[0]); int m = int.Parse(s[1]);
        int answer = 0;
        if(n >= 3) {
            if(m > 6) answer = m-2;
            else answer = Math.Min(4, m);
            
        } else {
            if(n==1) answer = 1;
            else answer = Math.Min(4, ((m-1)/2)+1);
        }
            
        WriteLine(answer);
    }
}

File: Beakjoon/Python/1783 병든 나이트.py
n,m = map(int, input().split())
answer = 0
if n == 1 :
    answer = 1
elif n == 2 :
    answer = min(4, (m-1)//2 + 1)
else :
    answer = m-2 if m > 6 else min(4, m) 

print(answer)
